File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/configurations/spring_boot/OverrideParameter.java
package com.github.alexandrecarlton.idea.settings.layout.configurations.spring_boot;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import org.immutables.value.Value;

@Value.Immutable
@JsonDeserialize(as = ImmutableOverrideParameter.class)
public interface OverrideParameter {
  String key();
  String value();
}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/configurations/spring_boot/OverrideParameter.kt
package com.github.alexandrecarlton.idea.settings.layout.configurations.spring_boot

import com.fasterxml.jackson.annotation.JsonProperty

data class OverrideParameter(

    @JsonProperty("Name")
    val name: String? = null,

    @JsonProperty("Value")
    val value: String? = null
)
